src/bin/nydus-image/main.rs

```diff
@@ -1778,15 +1797,80 @@ impl Command {
     fn get_backend(
         matches: &ArgMatches,
         blob_id: &str,
-    ) -> Result<Arc<dyn BlobBackend + Send + Sync>> {
-        let cfg_file = matches
-            .get_one::<String>("backend-config")
-            .context("missing backend-config argument")?;
-        let cfg = ConfigV2::from_file(cfg_file)?;
-        let backend_cfg = cfg.get_backend_config()?;
-        let backend = BlobFactory::new_backend(backend_cfg, blob_id)?;
-
-        Ok(backend)
+    ) -> Result<(Arc<ConfigV2>, Arc<dyn BlobBackend + Send + Sync>)> {
+        let config: Arc<ConfigV2>;
+        let backend: Arc<dyn BlobBackend + Send + Sync>;
+        if let Some(p) = matches.get_one::<String>("blob") {
+            config = Arc::new(ConfigV2::default());
+            backend = {
+                let blob_path = PathBuf::from(p);
+                let local_fs_conf = LocalFsConfig {
+                    blob_file: blob_path.to_str().unwrap().to_owned(),
+                    dir: Default::default(),
+                    alt_dirs: Default::default(),
+                };
+                let local_fs = LocalFs::new(&local_fs_conf, Some(blob_id))
+                    .with_context(|| format!("fail to create local backend for {:?}", blob_path))?;
+
+                Arc::new(local_fs)
+            };
+        } else if let Some(dir) = matches.get_one::<String>("blob-dir") {
+            config = Arc::new(ConfigV2::new_localfs("", dir)?);
+            backend = BlobFactory::new_backend(&config.backend.as_ref().unwrap(), blob_id)?;
+        } else if let Some(backend_type) = matches.get_one::<String>("backend-type") {
+            let content =
+                if let Some(backend_file) = matches.get_one::<String>("backend-config-file") {
+                    fs::read_to_string(backend_file).with_context(|| {
+                        format!("fail to read backend config file {:?}", backend_file)
+                    })?
+                } else if let Some(backend_config) = matches.get_one::<String>("backend-config") {
+                    backend_config.clone()
+                } else {
+                    bail!("--backend-config or --backend-config-file must be specified");
+                };
+
+            backend = if backend_type == "localfs" {
```

Comment:
Maybe refactor `BlobFactory::new_backend` or add a new `BlobFactory::new_backend_from` to create backend from config string, so that we can't miss the handle of the new backend types in future.